JASP-Engine/JASP/R/commonerrorcheck.R

```diff
@@ -456,4 +457,28 @@
     
   }
   return(result)
+}
+
+# Check if data set is variance-covariance matrix
+.checkVarCovMatrix <- function(dataset,nrow=TRUE,symm=TRUE,posdef=TRUE,...){
+
+  # as matrix:
+  dataMatrix <- as.matrix(dataset)
+  
+  # number of rows equal to number of columns?
+  if (nrow && nrow(dataset) != ncol(dataset)){
+    return(list(error=TRUE,reason="Dataset is not a square matrix"))
+  }
+  
+  # Symmetrical?
+  if (symm && !all(round(dataset,10) == t(round(dataset,10)))){
+    return(list(error=TRUE,reason="Matrix is not symmetrical"))
+  }
+  
+  # Positive-definite?
+  if (posdef && any(round(eigen(dataset)$values,10) < 0)){
```

Comment:
Yes, would it then maybe be more precise to change the reason to
reason="Matrix is not positive-semidefinite"?

# Positive-definite?
  if (posdef && any(round(eigen(dataset)$values,10) < 0)){
    return(list(error=TRUE,reason="Matrix is not positive-definite"))
  }